src/Suppressor.jl

```diff
@@ -37,11 +37,14 @@ macro suppress(block)
             _logger = current_logger()
         end
 
+        # Spelling out `with_logger(f, ...)` as a try/finally so we can do `@suppress using SpammyPkg`
+        t = current_task()
+        old_logstate = t.logstate
         try
-            Logging.with_logger(_logger) do
-                $(esc(block))
-            end
+            t.logstate = Base.CoreLogging.LogState(_logger)
+            $(esc(block))
         finally
+            t.logstate = old_logstate
```

Comment:
I think a nice alternative could be to write a small `@_with_logger(l)`  macro that expands to the same thing as what you have here, which would reduce your code duplication.

What do you think about that?

Reply:
Though i know there are some weirdnesses with calling one macro from another one.. :/ But i think it's not too bad? I don't remember the specifics though.

Reply:
Thanks! I can try but my macro-fu is not that strong.

Reply:
I just couldn't figure the macro in a macro out